structure/stack/stack_array.go

```diff
@@ -0,0 +1,89 @@
+// Stack Array
+// description: based on `geeksforgeeks` description Stack is a linear data structure which follows a particular order in which the operations are performed.
+//	The order may be LIFO(Last In First Out) or FILO(First In Last Out).
+// details:
+// 	Stack Data Structure : https://www.geeksforgeeks.org/stack-data-structure-introduction-program/
+// 	Stack (abstract data type) : https://en.wikipedia.org/wiki/Stack_(abstract_data_type)
+// author [Milad](https://github.com/miraddo)
+// see stacklinkedlist.go, stacklinkedlistwithlist.go, stack_test.go
+
+package stack
+
+/*
+The methods can also be implemented directly on the slice.
+```
+type StackArray[T any] []T
+```
+However, this exposes the underlaying storage (slice) outside the package.
+A struct is used instead, so that the underlying storage is not accessible outside the package.
+*/
+
+// StackArray is an implementation of stack with slice as underlying storage.
+// ```
+// stack := new(StackArray[int])
+// ```
+type StackArray[T any] struct {
```

Comment:
The word stack is already in the package name. The name `stack.StackArray` can be shortened to `stack.Array`.

Reply:
The `stack.Array` naming convention is the idiom in general. However, in this specific case, someone trying to learn the implementation in Go might be looking at this file alone (not with the perspective as a library code). Should I use the name `Array` itself or something like `ArrayStore` or `ArrayStack` (similar to `BtreeMap` - ordered map based on a B-Tree)

Reply:
> However, in this specific case, someone trying to learn the implementation in Go might be looking at this file alone (not with the perspective as a library code).

Personally, I think someone trying to learn an implementation in Go should also learn the idioms of Go so that they learn the idioms too 😄 

I think the name `Array` should suffice.